-diff --git a/3.4.1.patch b/3.4.1.patch
-index 3966fb6f..8e43c7cd 100644
---- a/3.4.1.patch
-+++ b/3.4.1.patch
-@@ -1,9 +1,89 @@
--diff --git a/3.4.1.patch b/3.4.1.patch
--new file mode 100644
--index 00000000..0beb5851
----- /dev/null
--+++ b/3.4.1.patch
--@@ -0,0 +1,85 @@
-+
+
 +++diff --git a/src/catch2/catch_assertion_result.cpp b/src/catch2/catch_assertion_result.cpp
 +++index dba86229..7bcaa2b8 100644
 +++--- a/src/catch2/catch_assertion_result.cpp
diff --git a/src/catch2/catch_assertion_result.cpp b/src/catch2/catch_assertion_result.cpp
index 986d1417..eca8cb32 100644
--- a/src/catch2/catch_assertion_result.cpp
+++ b/src/catch2/catch_assertion_result.cpp
@@ -1,3 +1,4 @@
+
 //              Copyright Catch2 Authors
 // Distributed under the Boost Software License, Version 1.0.
 //   (See accompanying file LICENSE.txt or copy at
@@ -31,10 +32,12 @@ namespace Catch {
         m_resultData( CATCH_MOVE(data) )
     {}
 
+    // Result was a success
     bool AssertionResult::succeeded() const {
         return Catch::isOk( m_resultData.resultType );
     }
 
+    // Result was a success, or failure is suppressed
     bool AssertionResult::isOk() const {
         return Catch::isOk( m_resultData.resultType ) || shouldSuppressFailure( m_info.resultDisposition );
     }
@@ -51,34 +54,8 @@ namespace Catch {
         return !m_resultData.message.empty();
     }
 
-    bool AssertionResult::hasCustomMessage() const {
-        return !m_resultData.customMessage.empty();
-    }
-
-    void AssertionResult::setCustomMessage() {
-        std::string customMessage;
-
-        if( hasExpression() ) {
-            customMessage += "assertion: ";
-            customMessage += getExpressionInMacro();
-        }
-
-        if( hasExpandedExpression() ) {
-            if( !customMessage.empty() ) {
-                customMessage += '\n';
-            }
-            customMessage += "expanded: ";
-            customMessage += getExpandedExpression();
-        }
-
-        if( customMessage.empty() && hasMessage() ) {
-            customMessage = static_cast<std::string>( getMessage() );
-        }
-
-        m_resultData.customMessage = CATCH_MOVE( customMessage );
-    }
-
     std::string AssertionResult::getExpression() const {
+        // Possibly overallocating by 3 characters should be basically free
         std::string expr; expr.reserve(m_info.capturedExpression.size() + 3);
         if (isFalseTest(m_info.resultDisposition)) {
             expr += "!(";
@@ -117,11 +94,6 @@ namespace Catch {
     StringRef AssertionResult::getMessage() const {
         return m_resultData.message;
     }
-
-    StringRef AssertionResult::getCustomMessage() const {
-        return m_resultData.customMessage;
-    }
-
     SourceLineInfo AssertionResult::getSourceInfo() const {
         return m_info.lineInfo;
     }
@@ -130,4 +102,16 @@ namespace Catch {
         return m_info.macroName;
     }
 
+    void AssertionResult::setCustomMessage(std::string const& msg) {
+        m_customMessage = msg;
+    }
+
+    std::string AssertionResult::getCustomMessage() const {
+        return m_customMessage;
+    }
+
+    bool AssertionResult::hasCustomMessage() const {
+        return !m_customMessage.empty();
+    }
+
 } // end namespace Catch
diff --git a/src/catch2/catch_assertion_result.hpp b/src/catch2/catch_assertion_result.hpp
index 47e01ded..36880735 100644
--- a/src/catch2/catch_assertion_result.hpp
+++ b/src/catch2/catch_assertion_result.hpp
@@ -1,3 +1,4 @@
+
 //              Copyright Catch2 Authors
 // Distributed under the Boost Software License, Version 1.0.
 //   (See accompanying file LICENSE.txt or copy at
@@ -24,7 +25,6 @@ namespace Catch {
         AssertionResultData( ResultWas::OfType _resultType, LazyExpression const& _lazyExpression );
 
         std::string message;
-        std::string customMessage;
         mutable std::string reconstructedExpression;
         LazyExpression lazyExpression;
         ResultWas::OfType resultType;
@@ -42,20 +42,22 @@ namespace Catch {
         ResultWas::OfType getResultType() const;
         bool hasExpression() const;
         bool hasMessage() const;
-        bool hasCustomMessage() const;
-        void setCustomMessage();
         std::string getExpression() const;
         std::string getExpressionInMacro() const;
         bool hasExpandedExpression() const;
         std::string getExpandedExpression() const;
         StringRef getMessage() const;
-        StringRef getCustomMessage() const;
         SourceLineInfo getSourceInfo() const;
         StringRef getTestMacroName() const;
 
+        void setCustomMessage(std::string const& msg);
+        std::string getCustomMessage() const;
+        bool hasCustomMessage() const;
+
     //protected:
         AssertionInfo m_info;
         AssertionResultData m_resultData;
+        std::string m_customMessage;
     };
 
 } // end namespace Catch
diff --git a/src/catch2/internal/catch_run_context.cpp b/src/catch2/internal/catch_run_context.cpp
index e9a47e12..789caa7c 100644
--- a/src/catch2/internal/catch_run_context.cpp
+++ b/src/catch2/internal/catch_run_context.cpp
@@ -849,8 +849,9 @@ namespace Catch {
 
         AssertionResult assertionResult{ info, CATCH_MOVE( data ) };
         assertionResult.m_resultData.lazyExpression.m_transientExpression = expr;
-        if ( resultType == ResultWas::ExpressionFailed ) {
-            assertionResult.setCustomMessage();
+
+        if (!assertionResult.isOk()) {
+            assertionResult.setCustomMessage("Custom thread-safe assertion failed info");
         }
 
         assertionEnded( CATCH_MOVE(assertionResult) );
diff --git a/src/catch2/reporters/catch_reporter_console.cpp b/src/catch2/reporters/catch_reporter_console.cpp
index 244fa86a..d9871c5c 100644
--- a/src/catch2/reporters/catch_reporter_console.cpp
+++ b/src/catch2/reporters/catch_reporter_console.cpp
@@ -58,11 +58,10 @@ public:
         case ResultWas::Ok:
             colour = Colour::Success;
             passOrFail = "PASSED"_sr;
-            if ( result.hasCustomMessage() )
-                messageLabel = "with custom message"_sr;
-            else if (messages.size() == 1)
+            //if( result.hasMessage() )
+            if (messages.size() == 1)
                 messageLabel = "with message"_sr;
-            else if (messages.size() > 1)
+            if (messages.size() > 1)
                 messageLabel = "with messages"_sr;
             break;
         case ResultWas::ExpressionFailed:
@@ -73,11 +72,9 @@ public:
                 colour = Colour::Error;
                 passOrFail = "FAILED"_sr;
             }
-            if ( result.hasCustomMessage() )
-                messageLabel = "with custom message"_sr;
-            else if (messages.size() == 1)
+            if (messages.size() == 1)
                 messageLabel = "with message"_sr;
-            else if (messages.size() > 1)
+            if (messages.size() > 1)
                 messageLabel = "with messages"_sr;
             break;
         case ResultWas::ThrewException:
@@ -139,6 +136,10 @@ public:
 
     void print() const {
         printSourceInfo();
+        if (result.hasCustomMessage()) {
+            stream << colourImpl->guardColour(Colour::Warning)
+                   << "  Custom Message: " << result.getCustomMessage() << '\n';
+        }
         if (stats.totals.assertions.total() > 0) {
             printResultType();
             printOriginalExpression();
@@ -156,18 +157,12 @@ private:
         }
     }
     void printOriginalExpression() const {
-        if ( result.hasCustomMessage() ) {
-            return;
-        }
         if (result.hasExpression()) {
             stream << colourImpl->guardColour( Colour::OriginalExpression )
                    << "  " << result.getExpressionInMacro() << '\n';
         }
     }
     void printReconstructedExpression() const {
-        if ( result.hasCustomMessage() ) {
-            return;
-        }
         if (result.hasExpandedExpression()) {
             stream << "with expansion:\n";
             stream << colourImpl->guardColour( Colour::ReconstructedExpression )
@@ -179,10 +174,8 @@ private:
     void printMessage() const {
         if (!messageLabel.empty())
             stream << messageLabel << ':' << '\n';
-        if ( result.hasCustomMessage() ) {
-            stream << TextFlow::Column( result.getCustomMessage() ).indent( 2 ) << '\n';
-        }
         for (auto const& msg : messages) {
+            // If this assertion is a warning ignore any INFO messages
             if (printInfoMessages || msg.type != ResultWas::Info)
                 stream << TextFlow::Column(msg.message).indent(2) << '\n';
         }
